src/main/java/seedu/duke/TransactionManager.java

```diff
@@ -253,6 +255,64 @@ public String getAllTransactionsOfMonth (String monthString) throws NoTransactio
         return stringToReturn;
     }
 
+    /**
+     * Method that performs validation on the LocalDate variable
+     * It ensures that the date doesn't have a day that is more than the number of days in its month
+     * The original dateString is required since LocalDate parse
+     * automatically reduces the day to the highest correct value if it exceeds the number of days in its month
+     *
+     * @param date       The date to be validated
+     * @param dateString The original dateString before being parsed
+     * @return The boolean representing whether the date is valid
+     */
+    private boolean isDateValid (LocalDate date, String dateString) {
+        Month month = date.getMonth();
+        int day;
+        try {
+            day = Integer.parseInt(dateString.substring(0, 2));
+        } catch (NumberFormatException e) {
+            assert false;
+            return false;
+        }
+
+        switch (month) {
```

Comment:
Hello, it may be good to use an array to store days in each month just to improve readability:

```java
int[] daysInMonth = {31, date.isLeapYear() ? 29 : 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

Reply:
Instead replaced the switch-case statements with a much shorter and cleaner implementation using a constant hashmap in date constants, DAYS_IN_MONTH

Reply:
Just looked over it, great change!